app/views/dashboard/autoyast.xml.erb

```diff
@@ -9,7 +9,12 @@
         <interpreter>shell</interpreter>
         <source><![CDATA[
         #!/bin/sh
-        echo "master: <%= @controller_node %>" > /etc/salt/minion.d/master.conf
+        MINION_ID=$(grep id /tmp/master-id | sed 's/id: //')
+        if [ -z "$MINION_ID" ]; then
+            UUID=$(uuidgen -r)
+            echo "id: $UUID" > /etc/salt/minion.d/master.conf      
+            echo "master: <%= @controller_node %>" > /etc/salt/minion.d/master.conf
+        fi
```

Comment:
This could be replaced completely by the following snippet, which is simpler, if we are talking about completely random `minion-ids` because it's generated on the rails application instead of the machine being installed:

```
        #!/bin/sh
        echo "id: <%= SecureRandom.hex %>" > /etc/salt/minion.d/master.conf
        echo "master: <%= @controller_node %>" >> /etc/salt/minion.d/master.conf
        ]]>
```

However I think we should try to get this ID from `/etc/machine-id` if that's possible at all. At least it would be good to explore this possibility.

This merge has some other implications in our salt states that we need to address. Maybe we should create a dedicated branch for this change (for example, on [salt states](https://github.com/kubic-project/salt/search?utf8=%E2%9C%93&q=grains&type=) we take as reference `grains['ip4_interfaces']['eth0'][0]` and I am not sure with this change if `grains['ip4_interfaces']['eth0'][0]` will be the hostname or the minion_id - probably the minion_id, so we nee to adapt some parts of our salt states)